From a63e9a21d8b4f40b53fbf1ea0d7aa7c6c257692b Mon Sep 17 00:00:00 2001
From: Sun Yuechi <sunyuechi@iscas.ac.cn>
Date: Thu, 25 Jun 2026 21:39:11 +0800
Subject: [PATCH 2/6] osd/ECOmapJournal: bind by reference when clearing omap
 keys

Signed-off-by: Sun Yuechi <sunyuechi@iscas.ac.cn>
---
 src/osd/ECOmapJournal.cc | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/osd/ECOmapJournal.cc b/src/osd/ECOmapJournal.cc
index c33af5ed70a..fef19671337 100644
--- a/src/osd/ECOmapJournal.cc
+++ b/src/osd/ECOmapJournal.cc
@@ -234,7 +234,7 @@ void ECOmapJournal::process_entries(const hobject_t &hoid) {
     // Clear omap if specified
     if (entry_iter->clear_omap) {
       // Mark all keys as removed
-      for (auto [_, value] : key_map[hoid]) {
+      for (auto &[_, value] : key_map[hoid]) {
         value.update_value(entry_iter->version, std::nullopt);
       }
 
-- 
2.54.0

