From 3632a1d1326b114bb35faedeaf770c1405314d8d Mon Sep 17 00:00:00 2001
From: Sun Yuechi <sunyuechi@iscas.ac.cn>
Date: Thu, 25 Jun 2026 21:39:11 +0800
Subject: [PATCH 4/6] osd/scheduler: add missing breaks in PGRecoveryMsg::run

Signed-off-by: Sun Yuechi <sunyuechi@iscas.ac.cn>
---
 src/osd/scheduler/OpSchedulerItem.cc | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/src/osd/scheduler/OpSchedulerItem.cc b/src/osd/scheduler/OpSchedulerItem.cc
index 7ed99a60098..f1d23c4e728 100644
--- a/src/osd/scheduler/OpSchedulerItem.cc
+++ b/src/osd/scheduler/OpSchedulerItem.cc
@@ -214,16 +214,22 @@ void PGRecoveryMsg::run(
   switch (op->get_req()->get_type()) {
   case MSG_OSD_PG_PUSH:
     osd->logger->tinc(l_osd_recovery_push_queue_lat, latency);
+    break;
   case MSG_OSD_PG_PUSH_REPLY:
     osd->logger->tinc(l_osd_recovery_push_reply_queue_lat, latency);
+    break;
   case MSG_OSD_PG_PULL:
     osd->logger->tinc(l_osd_recovery_pull_queue_lat, latency);
+    break;
   case MSG_OSD_PG_BACKFILL:
     osd->logger->tinc(l_osd_recovery_backfill_queue_lat, latency);
+    break;
   case MSG_OSD_PG_BACKFILL_REMOVE:
     osd->logger->tinc(l_osd_recovery_backfill_remove_queue_lat, latency);
+    break;
   case MSG_OSD_PG_SCAN:
     osd->logger->tinc(l_osd_recovery_scan_queue_lat, latency);
+    break;
   }
   osd->dequeue_op(pg, op, handle);
   pg->unlock();
-- 
2.54.0

