From e4c1d19f10979065d5838ce749c8622e46f3ecc9 Mon Sep 17 00:00:00 2001
From: Sun Yuechi <sunyuechi@iscas.ac.cn>
Date: Fri, 3 Apr 2026 01:03:50 +0800
Subject: [PATCH 3/3] src/xxHash: sync with upstream to enable RISC-V RVV
 optimizations

Signed-off-by: Sun Yuechi <sunyuechi@iscas.ac.cn>
---
 src/xxHash | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/xxHash b/src/xxHash
index bbb27a5efb8..0cee5d8c44d 160000
--- a/src/xxHash
+++ b/src/xxHash
@@ -1 +1 @@
-Subproject commit bbb27a5efb85b92a0486cf361a8635715a53f6ba
+Subproject commit 0cee5d8c44d32d7d2ce39096b63833a63d5211a8
-- 
2.54.0

